lumigator/backend/backend/api/routes/secrets.py

```diff
@@ -0,0 +1,46 @@
+from http import HTTPStatus
+
+from fastapi import APIRouter, status
+from lumigator_schemas.secrets import SecretGetRequest, SecretUploadRequest
+from starlette.requests import Request
+from starlette.responses import Response
+
+from backend.api.deps import SecretServiceDep
+from backend.services.exceptions.base_exceptions import ServiceError
+
+router = APIRouter()
+
+
+def secret_exception_mappings() -> dict[type[ServiceError], HTTPStatus]:
+    return {}
+
+
+@router.put(
+    "/{secret_name}",
+    status_code=status.HTTP_201_CREATED,
+    responses={
+        status.HTTP_201_CREATED: {"description": "Secret successfully created"},
+        status.HTTP_204_NO_CONTENT: {"description": "Secret successfully updated"},
+    },
+)
+def upload_secret(
+    service: SecretServiceDep,
+    secret_upload_request: SecretUploadRequest,
+    request: Request,
+    response: Response,
+    secret_name: str,
+) -> None:
```

Comment:
is this doing an `upsert`? it might be weird to get a `201 CREATED` or a `204 NO CONTENTz` when doing an update

I would expect this to return the updated/created item instead, and ideally have separate endpoints with different http methods for create vs update to be more RESTful

Is this a real `PUT` where the consumer needs to pass everything including `name` & `description` for example when changing a secret or do we want to allow more fine-grained partial updates via a `PATCH`?

Reply:
Being a secret, iirc this would be a hard put. You'll need to specify everything, plus won't get the value since it's, er, secret. Yes, it's an upsert, so If the value exists, it's overwritten, hence the 201/204.